mwdb/core/search/fields.py

```diff
@@ -253,20 +253,17 @@ def get_condition(self, expression: Expression, remainder: List[str]) -> Any:
         value = get_term_value(expression)
 
         def jsonpath_escape(field):
-            # Escapes field to be correctly represented in jsonpath
-            asterisk = ""
+            """Escapes field to be correctly represented in jsonpath"""
             # Escape all double quotes
             field = field.replace('"', '\\"')
-            if field.endswith("*"):
-                # If asterisk is escaped: remove the escape
-                if field[-2] == "\\" and field[-3] != "\\":
-                    # aaaa\* -> "aaaa*"
-                    field = field[:-2] + "*"
-                else:
-                    # aaaa* -> "aaaa"[*]
-                    asterisk = "[*]"
-                    field = field[:-1]
-            return f'"{field}"{asterisk}'
+            # Find trailing non-escaped asterisks
+            asterisk_r = re.search(r"(?<!\\)([*]+)$", field)
+            asterisk_levels = len(asterisk_r.group(0)) if asterisk_r else 0
+            field = field[:-asterisk_levels]
```

Comment:
With such an implementation we are not able to refer to individual fields.
This code causes that any field element without an asterisk change to empty string ("").